### PR TITLE
Greenhouse: switch to pd-ssd, drop --remount

### PR DESCRIPTION
**Pull request by BenTheElder**
- switch to PersistentVolumeClaim + GCE pd-ssd based StorageClass for storage
  - this gives us access to way more disk space, but we can still get comparable IOPs by having enough vCPU and enough GB of disk allocated (since it's actually network attached)
- drop --remount and associated code (we can supply the mount options on the PVC)
- rename deployment to greenhouse (not strictly necessary, but also helped switch to the new deployment while leaving the old one available w/ the local ssd etc.)
- update README to reflect new deployment details

/area greenhouse

### CODE REVIEW

**Comment by ixdy** [comment]
can you explain the reasons for this change?

**Comment by BenTheElder** [comment]
@ixdy *way* more disk space, but hopefully comparable iops specifically because it's a large amount of disk on a node with enough vCPU.

plus, we can now pass mount options natively (since we're using a PVC, not hostPath / local ssd), so we can drop the --remount hackery

File: greenhouse/gce-storage.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-ssd
+# we want to use a volume with strictatime,lazytime (and not noatime or relatime)
+# so that file access times *are* recorded but are lazily flushed to the disk
+# https://lwn.net/Articles/621046/
+# https://unix.stackexchange.com/questions/276858/why-is-ext4-filesystem-mounted-with-both-relatime-and-lazytime
+mountOptions: ["strictatime", "lazytime"]
+---
+# 3TB of SSD :-)
```
Comment (cjwagner) [inline greenhouse/gce-storage.yaml]: 😃

**Comment by cjwagner** [comment]
LGTM as well.

**Pull request merged**